subsys/bluetooth/services/dis.c

```diff
@@ -59,15 +107,133 @@ static ssize_t read_pnp_id(struct bt_conn *conn,
 /* Device Information Service Declaration */
 BT_GATT_SERVICE_DEFINE(dis_svc,
 	BT_GATT_PRIMARY_SERVICE(BT_UUID_DIS),
+
 	BT_GATT_CHARACTERISTIC(BT_UUID_DIS_MODEL_NUMBER,
 			       BT_GATT_CHRC_READ, BT_GATT_PERM_READ,
-			       read_str, NULL, CONFIG_BT_GATT_DIS_MODEL),
+			       read_str, NULL, BT_GATT_DIS_MODEL_REF),
 	BT_GATT_CHARACTERISTIC(BT_UUID_DIS_MANUFACTURER_NAME,
 			       BT_GATT_CHRC_READ, BT_GATT_PERM_READ,
-			       read_str, NULL, CONFIG_BT_GATT_DIS_MANUF),
+			       read_str, NULL, BT_GATT_DIS_MANUF_REF),
 #if CONFIG_BT_GATT_DIS_PNP
 	BT_GATT_CHARACTERISTIC(BT_UUID_DIS_PNP_ID,
 			       BT_GATT_CHRC_READ, BT_GATT_PERM_READ,
 			       read_pnp_id, NULL, &dis_pnp_id),
 #endif
+
+#if defined(CONFIG_BT_GATT_DIS_SERIAL_NUMBER)
+	BT_GATT_CHARACTERISTIC(BT_UUID_DIS_SERIAL_NUMBER,
+			       BT_GATT_CHRC_READ, BT_GATT_PERM_READ,
+			       read_str, NULL,
+			       BT_GATT_DIS_SERIAL_NUMBER_STR_REF),
+#endif
+#if defined(CONFIG_BT_GATT_DIS_FW_REV)
+	BT_GATT_CHARACTERISTIC(BT_UUID_DIS_FIRMWARE_REVISION,
+			       BT_GATT_CHRC_READ, BT_GATT_PERM_READ,
+			       read_str, NULL, BT_GATT_DIS_FW_REV_STR_REF),
+#endif
+#if defined(CONFIG_BT_GATT_DIS_HW_REV)
+	BT_GATT_CHARACTERISTIC(BT_UUID_DIS_HARDWARE_REVISION,
+			       BT_GATT_CHRC_READ, BT_GATT_PERM_READ,
+			       read_str, NULL, BT_GATT_DIS_HW_REV_STR_REF),
+#endif
+#if defined(CONFIG_BT_GATT_DIS_SW_REV)
+	BT_GATT_CHARACTERISTIC(BT_UUID_DIS_SOFTWARE_REVISION,
+			       BT_GATT_CHRC_READ, BT_GATT_PERM_READ,
+			       read_str, NULL, BT_GATT_DIS_SW_REV_STR_REF),
+#endif
+
 );
+
+#if defined(CONFIG_BT_SETTINGS) && defined(CONFIG_BT_GATT_DIS_SETTINGS)
+static int dis_set(int argc, char **argv, size_t len_rd, settings_read_cb read,
```

Comment:
rename read argument to read_cb, to avoid confusion?

Reply:
Done